src/components/Faucet/index.tsx

```diff
@@ -99,59 +107,91 @@ export default function FaucetPage({
         const address = await signer.getAddress()
-        setDid(address)
+        const network = await provider.getNetwork()
+        setAddress(address)
+        setNetwork(networkNameMap[network.chainId] || 'Unknown')
       } catch (error) {
         LoggerInstance.error(error)
       }
     }
 
-    fetchAddress()
+    fetchAddressAndNetwork()
 
-    if (didQueryString) {
-      setDid(didQueryString)
-      handleVerify()
+    if (window.ethereum) {
+      window.ethereum.on('accountsChanged', fetchAddressAndNetwork)
+      window.ethereum.on('chainChanged', fetchAddressAndNetwork)
+    }
+
+    return () => {
+      if (window.ethereum) {
+        window.ethereum.removeListener(
+          'accountsChanged',
+          fetchAddressAndNetwork
+        )
+        window.ethereum.removeListener('chainChanged', fetchAddressAndNetwork)
+      }
     }
-  }, [didQueryString, handleVerify])
+  }, [])
 
   return (
-    <div>
-      <form
-        className={styles.form}
-        onSubmit={async (e) => {
-          e.preventDefault()
-          handleSearchStart()
-        }}
-      >
-        <InputGroup>
-          <InputElement
-            className={styles.didInput}
-            label={label}
-            name="did"
-            onChange={(event) =>
-              setDid((event.target as HTMLInputElement).value)
-            }
-            placeholder={placeholder}
-            value={did}
-            readOnly
-          />
-          <Button
-            disabled={!did || isLoading}
-            style="primary"
-            size="small"
-            type="submit"
-          >
-            {isLoading ? <Loader /> : buttonLabel}
-          </Button>
-        </InputGroup>
+    <div className={styles.card}>
+      <h2 className={styles.title}>Welcome to the Pontus-X Faucet</h2>
+      <p className={styles.description}>
+        A faucet is a service that provides free tokens for testing purposes.
+        Known accounts can request 100 EUROe fee tokens and 1000 EUROe payment
+        tokens to use on the Pontus-X network. These tokens are available every
+        12 hours.
+      </p>
+      <div className={styles.instructions}>
+        <h3>How to Request Tokens:</h3>
+        <ol>
+          <li>Ensure you have a web3 wallet (e.g. MetaMask) connected.</li>
+          <li>
+            Your wallet address will be automatically detected and displayed
+            below.
+          </li>
+          <li>
+            Click the &quot;Get Tokens&quot; button to request your free tokens.
+          </li>
+          <li>Wait for a few seconds while the request is processed.</li>
+          <li>
+            You will receive a confirmation once the tokens are successfully
+            added to your wallet.
+          </li>
+        </ol>
+        <p>
+          <strong>Note:</strong> You can only request tokens once every 12
+          hours. If you encounter any issues, please try again later.
+        </p>
+      </div>
```

Comment:
suggestion: to be consistent in the repo we should move the content of this page to its own `content/.json` file
If this is blocking an earlier release we should create an issue to track progress for the task.

Reply:
This will be addressed in a subsequent pull request.